### PR TITLE
bpo-30533:Add function inspect.getmembers_static that does not call properties or dynamic properties.

### PR DESCRIPTION
**Pull request by hongweipeng**
https://bugs.python.org/issue30533

### CODE REVIEW

**Comment by kernc** [comment]
Also refs: https://bugs.python.org/issue38337
Closes: https://github.com/python/cpython/pull/16521

File: Lib/inspect.py
Comment (itamaro) [inline Lib/inspect.py]: if this function is converted to an internal implementation detail with non-optional `predicate` and `getter` args, need to either update the docstring to match the change, or drop the docstring altogether

File: Lib/inspect.py
Comment (itamaro) [inline Lib/inspect.py]: suggesting to change the docstring to be more explicit (and include some of the content as in getattr_static)

```suggestion
    """Return all members of an object as (name, value) pairs sorted by name
    without triggering dynamic lookup via the descriptor protocol,
    __getattr__ or __getattribute__. Optionally, only return members that
    satisfy a given predicate.
    
    Note: this function may not be able to retrieve all members
       that getmembers can fetch (like dynamically created attributes)
       and may find members that getmembers can't (like descriptors
       that raise AttributeError). It can also return descriptor objects
       instead of instance members in some cases.
    """
```

File: Misc/NEWS.d/next/Library/2020-06-16-18-00-56.bpo-30533.StL57t.rst
Comment (itamaro) [inline Misc/NEWS.d/next/Library/2020-06-16-18-00-56.bpo-30533.StL57t.rst]:
```suggestion
triggering dynamic lookup via the descriptor protocol. Patch by Weipeng Hong.
```

**Review by itamaro — COMMENTED** [review]
I noticed bpo-30533 has this patch attached.

I was able to apply this patch cleanly on latest main branch, and built and ran the `test_inspect` test.

Left a few comments around docstrings, but the functionality lgtm - would be great if a core dev could take a look :-)

**Review by DinoV — APPROVED** [review]
LGTM,!

**Pull request merged**
